toolkits/global/packages/global-article/__tests__/reading-companion.js

```diff
@@ -0,0 +1,394 @@
+		initReadingCompanion();		
+
+		const tabs = document.querySelector('.c-reading-companion__tabs');
+		const firstTab = tabs.querySelector('.c-reading-companion__tabs > li:first-child button');
+		const event = new KeyboardEvent('keydown', {'keyCode': 39});
+
+		tabs.dispatchEvent(event);
+
+		expect(firstTab.getAttribute('aria-selected')).toBe('true');
+
+		tabs.dispatchEvent(event);
+
+		expect(tabs.querySelector('[data-tab-target="figures"]').getAttribute('aria-selected')).toBe('true');
+	});
+
+	test('Should move the active tab to the one is clicked on', () => {
+		includeMockReferences();
+		includeMockFigures();
+		initReadingCompanion();
+
+		const referencesTab = document.querySelector('[data-tab-target="references"]');
+
+		referencesTab.click();
+		expect(referencesTab.getAttribute('aria-selected')).toBe("true");
+	});
+
+	test('Should update the active section on a new selection', () => {
+		addBackgroundSection();
+		initReadingCompanion();
+		emitter.emit('nav.section', 'Abs1', null);
+
+		expect(document.querySelector('#rc-sec-Abs1').classList.contains('c-reading-companion__section-item--active')).toBeTruthy();
+		
+		emitter.emit('nav.section', 'Sec1', 'Abs1');
+
+		expect(document.querySelector('#rc-sec-Abs1').classList.contains('c-reading-companion__section-item--active')).toBeFalsy();
+		expect(document.querySelector('#rc-sec-Sec1').classList.contains('c-reading-companion__section-item--active')).toBeTruthy();
+	});
+
+	test('Should switch to the figure tab and focus on the selected figure when clicking on a figure link', () => {
+		includeMockFigures();
+		addFigLinkSection();
+		initReadingCompanion();
+
+		const figureTab = document.querySelector('#tab-figures');
+		const selectedFigure = document.querySelector('#rc-Fig2');
+
+		selectedFigure.scrollIntoView = jest.fn();
+		selectedFigure.addEventListener = jest.fn();
+		emitter.emit('nav.figure', 'Fig2', null);
+
+		expect(figureTab.getAttribute('aria-selected')).toBe('true');
+		expect(figureTab.classList.contains('c-reading-companion__tab--active')).toBeTruthy();
+		expect(selectedFigure.getAttribute('tabindex')).toBe('-1');
+		expect(selectedFigure.classList.contains('c-reading-companion--highlighted')).toBeTruthy();
+		expect(selectedFigure.addEventListener).toHaveBeenCalled();
+		expect(selectedFigure.scrollIntoView).toHaveBeenCalledWith({block: 'start'});
+	});
+
+	test('Should switch to the references tab and focus on the selected reference when clicking on a reference link', () => {
+		// TODO: add more tests for insertReturnButton function
+		includeMockReferences();
+		initReadingCompanion();
+
+		const referencesTab = document.querySelector('#tab-references');
+		const selectedReference = document.querySelector('#rc-ref-CR1');
+
+		selectedReference.scrollIntoView = jest.fn();
+		selectedReference.addEventListener = jest.fn();
+		emitter.emit('nav.reference', 'ref-CR1', null);
+
+		expect(referencesTab.getAttribute('aria-selected')).toBe('true');
+	});
+});
```

Comment:
I'm wondering if because the test file is fairly big, I would quite like to pull out the HTML into separate files, and maybe some of the functions that are only called once, could be put inline..

Reply:
the file tested is actually big as well. We're planning to split it out in separate things as it is doing too many things.

You're right about adding that mocked HTML inline once and avoid calling those mock functions again and again. I actually thought to do it but I was a bit sick of this component and wanted to finish it haha! :stuck_out_tongue_closed_eye but yeah! you're right. Let's do the right thing :)